bcda/main.go

```diff
@@ -379,3 +390,45 @@ func getEnvInt(varName string, defaultVal int) int {
 	}
 	return defaultVal
 }
+
+func removeExpired(hrThreshold int) {
+	log.Info("Cleaning up expired jobs...")
+	db := database.GetGORMDbConnection()
+	defer db.Close()
+
+	var jobs []models.Job
+	err := db.Find(&jobs, "status = ?", "Completed").Error
+	if err != nil {
+		log.Error(err)
+	}
+
+	expDir := os.Getenv("FHIR_EXPIRED_DIR")
+	if _, err = os.Stat(expDir); os.IsNotExist(err) {
+		err = os.MkdirAll(expDir, os.ModePerm)
+		if err != nil {
+			log.Error(err)
+		}
+	}
+
+	for _, j := range jobs {
+		t := j.CreatedAt
+		elapsed := time.Since(t).Hours()
+		if int(elapsed) >= hrThreshold {
+
+			id := int(j.ID)
+			jobDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_PAYLOAD_DIR"), id)
+			expDir = fmt.Sprintf("%s/%d", os.Getenv("FHIR_EXPIRED_DIR"), id)
+
+			err = os.Rename(jobDir, expDir)
+			if err != nil {
+				log.Error(err)
```

Comment:
In order for this to actually be robust in the case of failures, I think you can't proceed past here in the case of errors, because you're still marking the cleanup as done in the DB even when the rename failed. Two suggestions:

1. Do whatever the Golang equivalent of `continue;` is after you log the error to skip the rest of this iteration.
2. After fixing that, if there's something up with the filesystem that will always prevent a `rename` from succeeding, then some things will stay marked as ready-to-expire in the DB forever.
    * You should add monitoring to ensure that things actually are being cleaned up in the time period we expect them to be. Something like a CloudWatch alarm or somesuch, so that folks are actually prompted to go investigate it.
        * If it's not already covered by something else, e.g. an alarm that fires for every error log entry.
        * Could be done as part of a separate ticket, but you should reference that ticket here or in JIRA.
        * (I didn't check JIRA, so apologies if that's already there.)

(And as always, apologies if I'm just missing something obvious.)

Reply:
1) Yep!! I see your point about the continue.
2) Let's say something does happen where we fail on this, the next time this code runs it will cleanup the "ready-to-expire" jobs on the next go-round assuming no real outstanding issues. But to your point if we do have a **real issue** and there's something up with the filesystem, (then we obviously having larger problems), and something like a CloudWatch alarm could definitely be implemented. I'll search for a corresponding ticket to link to this or create one if I can't find it.

Reply:
@tbellj I don't think there's an existing ticket for monitoring this. Would you mind creating one to instrument this process so that we can be notified if it is not working properly?